paasta_tools/autoscaling/autoscaling_cluster_lib.py

```diff
@@ -313,6 +337,10 @@ def set_spot_fleet_request_capacity(sfr_id, capacity, dry_run, region=None):
                 state = get_sfr(sfr_id, region=region)['SpotFleetRequestState']
                 if state == 'active':
                     break
+                if state == 'cancelled_running':
+                    log.info("Not updating target capacity because this is a cancelled SFR, "
+                             "we are just draining and killing the instances")
+                    return
                 log.debug("SFR {0} in state {1}, waiting for state: active".format(sfr_id, state))
                 log.debug("Sleep 5 seconds")
                 time.sleep(5)
```

Comment:
I think it's okay? If we reach the timeout then we end up raising an exception before we get to L353. If it is cancelled_running then we return so we don't reach any set calls.

The reasoning you state is correct: you can't set a TargetCapacity on a cancelled SFR. And that's because AWS won't launch instances for cancelled SFRs so a target is meaningless.